### PR TITLE
fix for #1897 and minor example fixes

### PR DESCRIPTION
**Pull request by mlorek**
fix for #1897

### CODE REVIEW

**Comment by KvanTTT** [comment]
Could you also please that minor issue? https://github.com/antlr/grammars-v4/pull/2464#pullrequestreview-842299998

**Comment by mlorek** [comment]
> Could you also please that minor issue? [#2464 (review)](https://github.com/antlr/grammars-v4/pull/2464#pullrequestreview-842299998)

already there

**Comment by KvanTTT** [comment]
`yield` is still missing. Also, put `Java 17` to `Supported Java Version`: https://github.com/antlr/grammars-v4/tree/master/java/java#supported-java-versions

**Comment by mlorek** [comment]
> `yield` is still missing (or maybe there are some problems after adding?). Also, put `Java 17` to `Supported Java Version`: https://github.com/antlr/grammars-v4/tree/master/java/java#supported-java-versions

my bad

File: java/java/JavaParser.g4 [hunk 1]
```diff
@@ -267,7 +267,13 @@ qualifiedNameList
     ;
 
 formalParameters
-    : '(' formalParameterList? ')'
+    : '(' receiverParameter? ')'
+    | '(' receiverParameter (',' formalParameterList)? ')'
+    | '(' formalParameterList? ')'
```
Comment (KvanTTT) [inline java/java/JavaParser.g4:272]: Maybe merge (a bit more optimal)?

```suggestion
    : '(' ( receiverParameter?
          | receiverParameter (',' formalParameterList)?
          | formalParameterList?
       ) ')'
```

**Pull request merged**